script.js

```diff
@@ -0,0 +1,449 @@
+        recipeImage.src = obj.pictureLink;
+        recipeImage.alt = obj.title;
+        recipeCard.appendChild(recipeImage)    
+
+        const recipePitch = document.createElement('p');
+        recipePitch.innerText = obj.pitch;
+        recipeCard.appendChild(recipePitch);
+
+        
+
+        parent.appendChild(recipeCard)
+
+    //making cards expand on click
+            recipeCard.addEventListener("click", ()=>{
+                if (!recipeCard.querySelector('.ingredients') && !recipeCard.querySelector('.description')) {
+                    const ingredientsList = document.createElement('ul');
+                    ingredientsList.classList.add('ingredients'); 
+                    obj.ingredients.forEach((ingredient) => {
+                        const listItem = document.createElement('li');
+                        listItem.innerText = ingredient.amount > 0 
+                            ? `${ingredient.amount || ""} ${ingredient.unit || ""} of ${ingredient.name}`
+                            : `${ingredient.name}: to taste`;
+                        ingredientsList.appendChild(listItem);
+                    });
+                    recipeCard.appendChild(ingredientsList);
+
+                    const recipeDescription = document.createElement('p');
+                    recipeDescription.innerText = `Description: ${obj.description}`;
+                    recipeDescription.classList.add('description'); 
+                    recipeCard.appendChild(recipeDescription);
+
+                    const collapseButton = document.createElement('button');
+                    collapseButton.innerText = 'Collapse';
+                    collapseButton.classList.add('collapse-button'); 
+                    recipeCard.appendChild(collapseButton);
+
+                    // Make a card collapse
+                    collapseButton.addEventListener("click", (e) => {
+                        e.stopPropagation(); // Prevent triggering the card click event
+                        ingredientsList.remove();
+                        recipeDescription.remove();
+                        collapseButton.remove();
+                    });
+        }
+            })
+
+    
+    })
+}
+
+//rendering all recipes
+renderRecipeCard(recipesContainer, allRecipes);
+
+//find recipes by number of ingredients
+const getRecipeByIngredientNbr = (num)=>{
+    const foundRecipes = [];
+    allRecipes.forEach((obj)=>{
+        if(obj.ingredients.length===num) {
+            foundRecipes.push(obj);
+        }
+    })
+    return foundRecipes;
+}
+
+
+ingredientSearchBtn.addEventListener("click", (e)=> {
+    const num = Math.sqrt((parseInt(ingredientNumber.value) ** 2)); // because I'm lazy I am getting the square root of the square of the input, to make sure it's not a negative value
+    result = getRecipeByIngredientNbr(num);
+    renderRecipeCard(recipesSearchResults, result);
+
+})
+
+
+
+// finding recipes by keywords
```

Comment:
Add a more detailed comment here explaining where do you search about the keyword